Makefile

```diff
@@ -20,9 +20,5 @@ doc: ## Generate documentation
 
 doc-internal: ## Generate documentation with private items
 	@cargo rustdoc --lib -- --document-private-items
-
-publish-doc: ### Publish the documentation as github pages
```

Comment:
Why you removed the `publish-doc` from the Makefile, instead of calling it from `travis`? The Makefile is meant to be generic, so it doesn't need hard coded repo name, and it would be nice if travis configuration file can be too – so that we can have a common one to move around.

Reply:
The common way of doing this that we have now on the majority of our repos is basically do it on the way on which is done in this PR.

Since then, no GH Token has to be disclosed and also, no one needs this functionality on the makefile appart from Travis. So instead of adding this option here, It's better to apply the logic directly in the Travis pipeline.

Reply:
I forgot to remove the "pending" status of this comment: me and @CPerezz had a call where we sorted out everything; so we're going to land this PR as is, and I filed a issue to remove the publish-doc logic from Travis (see: #20) and make it a bit more generic.